project-koku/koku
# [COST-3539] masu task process daily partitioned OCP on Cloud Parquet

## Jira Ticket

[COST-3539](https://issues.redhat.com/browse/COST-3539)

## Description

This change will update the process OpenShift on cloud masu endpoint to handle providers that are partitioned daily (currently OCP on GCP). The task is separated from the non-daily task to make a potential future transition of all providers to daily partitions easier.

## Testing

1. Checkout Branch
2. Restart Koku
3. Load data with `make create-test-customer` and `make load-test-customer-data test_source=GCP`
4. (Optional): either use the trino CLI to gather the data for a specific day that will be resummarized or write some trino SQL to view the information. Trino CLI example:
```
trino --server localhost:8080 --catalog hive --schema org1234567 --user admin --debug --output-format=CSV_HEADER_UNQUOTED --execute "select * from gcp_openshift_daily where month='04' and day='01';" > gcpday.csv
```
5. After data has loaded, find your provider_uuid and replace it in the masu call:
6. http://localhost:5042/api/cost-management/v1/report/process/openshift_on_cloud/?provider_uuid=96ea7bb3-e856-460a-903d-e56bfc760a6c&schema=org1234567&start_date=2023-03-20&end_date=2023-04-10
7. You can watch the logs and see the days processing by, example: 
```
koku-koku-worker-1  | [2023-04-17 10:25:04,038] INFO 1c9b0d24-7f76-4f2e-9e49-4be88e18368b Processing OpenShift on GCP to parquet for Openshift source 24918501-f97f-4a75-8e00-49d14eed012b
koku-koku-worker-1  |   Start date: 2023-04-09
```
8. After the data has resummarized, gather the csv data again to compare to the previous output and ensure they match:
```
trino --server localhost:8080 --catalog hive --schema org1234567 --user admin --debug --output-format=CSV_HEADER_UNQUOTED --execute "select * from gcp_openshift_daily where month='04' and day='01';" > gcpday_post_summary.csv
```

## Notes

...
